### ISSUE TITLE
Is data should be function in a component?

### ISSUE THREAD

**Issue by shellvip** [KingPixil/moon#56]
```
	<script src="https://unpkg.com/moonjs"></script>
	<div id="app1">
		<simple-counter></simple-counter>
		<simple-counter></simple-counter>
		<simple-counter></simple-counter>
	</div>
	<script>
		Moon.component('simple-counter', {
			template: '<button m-on:click="add">{{ counter }}</button>',
			data: {
				counter: 0
			},
			methods: {
				add: function () {
					var v = this.get('counter');
					this.set('counter', v + 1);
				}
			}
		})
		var app = new Moon({
			el: "#app1"
		})
	</script>
```

the result is strange.

**Comment by KingPixil** [comment]
You're right, since data is passed by reference. This will be fixed in v0.11.0.

**Comment by KingPixil** [comment]
Fixed in 8f053217, coming out in v0.11.0
